tests/filename/filenametest.cpp

```diff
@@ -710,8 +712,24 @@ TEST_CASE("wxFileName::Exists", "[filename]")
     // These files are only guaranteed to exist under Linux.
     // No need for wxFILE_EXISTS_NO_FOLLOW here; wxFILE_EXISTS_SYMLINK implies it
     CHECK( wxFileName::Exists("/proc/self", wxFILE_EXISTS_SYMLINK) );
-    CHECK( wxFileName::Exists("/dev/log", wxFILE_EXISTS_SOCKET) );
 #endif // __LINUX__
+    wxString name = dirTemp.GetPath() + "/socktmpdirXXXXXX";
+    wxString socktempdir = wxString::From8BitData(mkdtemp(name.char_str()));
+    wxString sock = socktempdir + "/socket";
+    struct sockaddr_un addr;
+    int fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (fd)
+    {
+        memset(&addr, 0, sizeof(struct sockaddr_un));
+        addr.sun_family = AF_UNIX;
+        strncpy(addr.sun_path, sock.c_str(), sizeof(addr.sun_path) - 1);
+        if (!bind(fd, (struct sockaddr *) &addr, sizeof(struct sockaddr_un)))
+        {
+            wxON_BLOCK_EXIT1(wxRemoveFile, sock);
+            CHECK( wxFileName::Exists(sock, wxFILE_EXISTS_SOCKET) );
+        }
+    }
+    wxFileName::Rmdir(socktempdir);
```

Comment:
This should be done in `wxON_BLOCK_EXIT` too, shouldn't it?

Unless we decide to add `wxTempDir` class (which would do it in its dtor).

Reply:
Yeah a `wxTempDir` class would be really useful.  I was surprised it didn't exist.

Reply:
Allright, my OCD convinced me to add a `wxTempDir` class...

Reply:
Yay for the OCD!

But there doesn't seem to be such class in this PR, have you forgotten to push it here perhaps? Or do you plan to change this later, i.e. should I merge this already?

Thanks!

Reply:
I plan to do it later (probably not until Sunday or Monday).  You can merge this if you think it's ready, or wait.

One question though - do you think the new class should be part of the public API, or just an internal class for wx use?

Reply:
OK, I'll merge this one already, thanks.

And I do believe that `wxTempDir` would be useful to have in the public API. TIA!